### DISCUSSION TITLE
Appending generated provider to an existing providers.tf file

### DISCUSSION THREAD

**Discussion by lmilbaum** [gruntwork-io/knowledge-base#771] [How do I]
Each of my modules have a list of providers tracked in providers.tf file. 
I'd like the aws provider to be generated by terragrunt with the generate block.
How would I go about it?

---

<ins datetime="2023-09-24T18:24:54Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110468">Tracked in ticket #110468</a></p>
</ins>

**Comment by denis256** [comment]
Hello,
AFAIK this is not supported in Terragrunt, with `generate` block it is not possible to replace/append content in `tf` files only to generate new files

References:
https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#generate
